src/main/java/com/flippingutilities/controller/RecipeHandler.java

```diff
@@ -0,0 +1,321 @@
+        recipeFlipGroup.addRecipeFlip(recipeFlip);
+        recipeFlipGroups.add(recipeFlipGroup);
+    }
+
+    public List<RecipeFlipGroup> sortRecipeFlipGroups(List<RecipeFlipGroup> items, SORT selectedSort, Instant startOfInterval) {
+        List<RecipeFlipGroup> result = new ArrayList<>(items);
+
+        if (selectedSort == null || result.isEmpty()) {
+            return result;
+        }
+
+        switch (selectedSort) {
+            case TIME:
+                result.sort(Comparator.comparing(RecipeFlipGroup::getLatestFlipTime));
+                break;
+            case FLIP_COUNT:
+                result.sort(Comparator.comparing(group -> {
+                    List<RecipeFlip> flips = group.getFlipsInInterval(startOfInterval);
+                    return flips.stream().mapToInt(rf -> rf.getRecipeCountMade(group.getRecipe())).sum();
+                }));
+                break;
+            case TOTAL_PROFIT:
+                result.sort(Comparator.comparing(group -> {
+                    List<RecipeFlip> flips = group.getFlipsInInterval(startOfInterval);
+                    return flips.stream().mapToLong(RecipeFlip::getProfit).sum();
+                }));
+                break;
+            case PROFIT_EACH:
+                result.sort(Comparator.comparing(group -> {
+                    List<RecipeFlip> flips = group.getFlipsInInterval(startOfInterval);
+                    long totalProfit = flips.stream().mapToLong(RecipeFlip::getProfit).sum();
+                    long totalRecipesMade = flips.stream().mapToInt(rf -> rf.getRecipeCountMade(group.getRecipe())).sum();
+                    return totalProfit/totalRecipesMade;
+                }));
+                break;
+            case ROI:
+                result.sort(Comparator.comparing(group -> {
+                    List<RecipeFlip> flips = group.getFlipsInInterval(startOfInterval);
+                    long totalProfit = flips.stream().mapToLong(RecipeFlip::getProfit).sum();
+                    long totalExpense = flips.stream().mapToLong(RecipeFlip::getExpense).sum();
+                    return (float) totalProfit / totalExpense * 100;
+                }));
+                break;
+        }
+        Collections.reverse(result);
+        return result;
+    }
+
+    public void deleteInvalidRecipeFlips(List<OfferEvent> offers, List<RecipeFlipGroup> recipeFlipGroups) {
+        if (offers.isEmpty()) {
+            return;
+        }
+        int itemId = offers.get(0).getItemId();
+        //get all the rfgs containing recipe flips that the offers could possibly be referenced in
+        recipeFlipGroups.stream().filter(rfg -> rfg.isInGroup(itemId)).forEach(rfg -> rfg.deleteFlipsWithDeletedOffers(offers));
+    }
+
+    private Optional<List<Recipe>> loadRecipes() {
+        Request request = new Request.Builder()
+                .url("https://raw.githubusercontent.com/Flipping-Utilities/osrs-datasets/master/recipes.json")
+                .build();
```

Comment:
Should we cache this locally for a day or so?

Reply:
I guess we could, but it is only requested at plugin startup, which shouldn't be too bad.